redfish-core/lib/assembly.hpp

```diff
@@ -321,6 +332,80 @@ inline void setAssemblylocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+
+	// Handle special case for tod_battery assembly OEM ReadyToRemove property
+	// NOTE: The following method for the special case of the tod_battery
+	// ReadyToRemove property only works when there is only ONE adcsensor
+	// handled by the adcsensor application.
+	if (sdbusplus::message::object_path(assembly).filename() == "EBMC")
+	{
+	    std::optional<nlohmann::json> oem;
+	    if (!json_util::readJson(req, asyncResp->res, "Oem", oem))
+            {
+	        return;
+	    }
+	    if (oem)
+	    {
+	        std::optional<nlohmann::json> openbmc;
+		if (!json_util::readJson(*oem, asyncResp->res, "OpenBMC", openbmc))
+		{
+		    BMCWEB_LOG_ERROR << "Property Missing ";
+		    messages::propertyMissing(asyncResp->res, "OpenBMC");
+		    return;
+		}
+		if (openbmc)
+		{
+		    std::optional<bool> readytoremove;
+		    if (!json_util::readJson(*openbmc, asyncResp->res,
+					    "ReadyToRemove", readytoremove))
+		    {
+		        BMCWEB_LOG_ERROR << "Property Missing ";
+			messages::propertyMissing(asyncResp->res, "ReadyToRemove");
+			return;
+		    }
+		    
+                    // SetUp call to systemd to start or stop adcsensor
+		    const std::string service = {"adcsensor.service"};
+		    auto bus = sdbusplus::bus::new_default();
+
+		    if (readytoremove)
+		    {
+		        // Call systemd to turn off adcsensor
+		        auto method = bus.new_method_call("org.freedesktop.systemd1",
```

Comment:
This is a blocking call right? 
We should follow https://github.com/openbmc/bmcweb/search?q=systemd1